src/spikeinterface/core/generate.py

```diff
@@ -1208,8 +1209,11 @@ def get_traces(
         else:
             traces = np.zeros([end_frame - start_frame, n_channels], dtype=self.dtype)
 
-        start = np.searchsorted(self.spike_vector["sample_index"], start_frame - self.templates.shape[1], side="left")
-        end = np.searchsorted(self.spike_vector["sample_index"], end_frame + self.templates.shape[1], side="right")
+        start, end = np.searchsorted(
+            self.spike_vector["sample_index"],
+            [start_frame - self.templates.shape[1], end_frame + self.templates.shape[1] + 1],
+            side="left",
+        )
 
```

Comment:
```suggestion

         start = np.searchsorted(self.spike_vector["sample_index"], start_frame - self.templates.shape[1], side="left")
         end = np.searchsorted(self.spike_vector["sample_index"], end_frame + self.templates.shape[1], side="right")
        )

```

For this one is a left search and one is a right search. Maybe this is part of the problem?

Reply:
But not sure this is causing the exact failure?

Reply:
The +1 should handle the left/right discrepancy I guess, but I'll try to find why we have failures

Reply:
If the +1 is at the edge of the array, search sorted returns length of the array. So this would fail on edge cases where someone wants that edge.

Reply:
Just as super simple example: if our end frame has a value of 4 and we add one to switch it to a left search this could happen:

```
np.searchsorted([1,2,4], 5)
>>> 3
```

Reply:
My fault, I change some line while copy/pasting function calls.

Reply:
Right. Maybe I was too quick handling these right cases. Let's wait for test first, then otherwise I could revert old behaviour when right insertion are needed

Reply:
@yger please go back to the previous behavior do not change the right/left it is sometimes super complicated to debug!